internal/samsahai/activepromotion/demote.go

```diff
@@ -39,7 +39,16 @@ func (c *controller) demoteActiveEnvironment(ctx context.Context, atpComp *s2hv1
 				"team", teamName, "namespace", prevNs)
 			atpComp.Status.SetCondition(s2hv1.ActivePromotionCondActiveDemoted, corev1.ConditionFalse,
 				"Failed to demote active environment, active environment has been deleted")
+
+			if *atpComp.Spec.NoDowntimeGuarantee {
```

Comment:
please also check atpComp.Spec.NoDowntimeGuarantee must not be nil, just in case